AP_RangeFinder: Set default I2C address.

muramura opened the pull request:
I change the default I2C address of the device to always generate when the I2C address setting is 0.
I find it cumbersome to find the I2C address from the datasheet when using the device.
I want to use the device more easily.

lucasdemarchi commented:
0 is a valid address. Maybe the default should be, say, 128, and we document that 128 means "automatic"/device-specific?

muramura commented:
@lucasdemarchi san Comments thanks.

I don't use the address field for this device.
I specify the default address for this device, like the barometric sensor class.

I2C spec:
![Screenshot from 2019-11-16 07-05-37](https://user-images.githubusercontent.com/646194/68978956-8bd25280-083f-11ea-9b26-a80278303283.png)

lucasdemarchi commented:
I think you didn't understand. You are checking if the value is 0 and then providing a device-specific default address. The problem is that 0 is a valid I2C address according to the I2C spec.

So I'm saying to use another value as the param default (like say 128 since we don't support 10-bit addressing) and document this behavior.

muramura commented:
@lucasdemarchi san. Re comments thanks..

I understood that the I2C address is 0-127.
I will use the default I2C address of the device if 128 is set.
I add to this device's WIKI that if I2C addressing is set to 128, the device's default address will be used.

lucasdemarchi commented:
> I will use the default I2C address of the device if -1 is set.

as long as -1 is the default param value, ok.

> I add to this device's WIKI that if I2C addressing is set to -1, the device's default address will be used.

should be in the param documentation

> I set the value of the range finder address variable to −1 because the variable type is AP_INT8.

makes sense

So in addition to your updated changes you should have:

```diff
diff --git a/libraries/AP_RangeFinder/AP_RangeFinder_Params.cpp b/libraries/AP_RangeFinder/AP_RangeFinder_Params.cpp
index e00699b0a6..21db2aa6c4 100644
--- a/libraries/AP_RangeFinder/AP_RangeFinder_Params.cpp
+++ b/libraries/AP_RangeFinder/AP_RangeFinder_Params.cpp
@@ -91,11 +91,11 @@ const AP_Param::GroupInfo AP_RangeFinder_Params::var_info[] = {
 
     // @Param: ADDR
     // @DisplayName: Bus address of sensor
-    // @Description: This sets the bus address of the sensor, where applicable. Used for the I2C and UAVCAN sensors to allow for multiple sensors on different addresses. A value of 0 disables the sensor.
-    // @Range: 0 127
+    // @Description: This sets the bus address of the sensor, where applicable. Used for the I2C and UAVCAN sensors to allow for multiple sensors on different addresses. If set to -1, then the device-specific default address will be used
+    // @Range: -1 127
     // @Increment: 1
     // @User: Standard
-    AP_GROUPINFO("ADDR", 23, AP_RangeFinder_Params, address, 0),
+    AP_GROUPINFO("ADDR", 23, AP_RangeFinder_Params, address, -1),
 
     // @Param: POS_X
     // @DisplayName:  X position offset
```

The previous description saying that 0 is to disable should be removed because: 1) it's not respected by most of the I2C sensors and 2) 0 is a valid address.  However you will have to define a default address for each of the sensors. For TFMiniPlus that is 0x10.

muramura commented:
@lucasdemarchi san. Comment and Sample code, thanks.

I will reflect the comments.

muramura commented:
@lucasdemarchi san. 

I changed the value of the I2C address in the configuration parameter from 0 to -1.
I also changed other I2C devices with this change.
